python/hail/expr.py

```diff
@@ -26,6 +27,7 @@ class Type(object):
 
     def __init__(self, jtype):
         self._jtype = jtype
+        self._history = None
```

Comment:
You record history on `Type`?  I never would have thought of doing that!  How does this show up practically in the history?